logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/receiver/conversation/MLSWelcomeEventHandler.kt

```diff
@@ -38,19 +40,28 @@ internal class MLSWelcomeEventHandlerImpl(
     val conversationDAO: ConversationDAO
 ) : MLSWelcomeEventHandler {
     override suspend fun handle(event: Event.Conversation.MLSWelcome) {
-        mlsClientProvider.getMLSClient().flatMap { client ->
-            wrapMLSRequest { client.processWelcomeMessage(event.message.decodeBase64Bytes()) }
-                .flatMap { groupID ->
-                    kaliumLogger.i("Created conversation from welcome message (groupID = $groupID)")
+        val logMap = mutableMapOf<String, Any?>(
+            "event" to event.toLogMap(),
+        )
 
-                    wrapStorageRequest {
-                        if (conversationDAO.getConversationByGroupID(groupID).first() != null) {
-                            // Welcome arrived after the conversation create event, updating existing conversation.
-                            conversationDAO.updateConversationGroupState(ConversationEntity.GroupState.ESTABLISHED, groupID)
-                            kaliumLogger.i("Updated conversation from welcome message (groupID = $groupID)")
+        mlsClientProvider
+            .getMLSClient()
+            .flatMap { client ->
+                wrapMLSRequest { client.processWelcomeMessage(event.message.decodeBase64Bytes()) }
+                    .flatMap { groupID ->
+
+                        logMap["info"] = "Created conversation from welcome message"
+                        logMap["groupId"] = groupID.obfuscateId()
+
+                        wrapStorageRequest {
+                            if (conversationDAO.getConversationByGroupID(groupID).first() != null) {
+                                // Welcome arrived after the conversation create event, updating existing conversation.
+                                conversationDAO.updateConversationGroupState(ConversationEntity.GroupState.ESTABLISHED, groupID)
+                                logMap["info"] = "Updated conversation from welcome message"
+                            }
+                            kaliumLogger.i("Success Handling Event: ${logMap.toJsonElement()}")
                         }
                     }
```

Comment:
```suggestion
                        }
                        kaliumLogger.i("Success Handling Event: ${logMap.toJsonElement()}")
                    }
```

Let's move log outside the `wrapStorageRequest`? Same behaviour just looks weird to have it inside.

Reply:
That would change the behaviour

Reply:
Ah right the return changes, my bad